sdk/azidentity/azidentity_test.go

```diff
@@ -23,10 +28,114 @@ const (
 
 // constants for this file
 const (
-	envHostString    = "https://mock.com/"
-	customHostString = "https://custommock.com/"
+	envHostString              = "https://mock.com/"
+	customHostString           = "https://custommock.com/"
+	getTenantDiscoveryResponse = `{
```

Comment:
I'd probably put this and the validation function in `client_certificate_credential_test.go`. This file's more about high-level tests and broadly useful helpers
```suggestion
	tenantDiscoveryResponse = `{
```